File: crescer/java/trabalhos/tcc/api/src/main/java/br/com/cwi/crescer/api/service/security/GetUserSecurityService.java
package br.com.cwi.crescer.api.service.security;

import br.com.cwi.crescer.api.model.UserAccount;
import br.com.cwi.crescer.api.repository.UserAccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class GetUserSecurityService implements UserDetailsService {

    @Autowired
    private UserAccountRepository userAccountRepository;

    @Override
    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
        UserAccount userAccount = userAccountRepository.findByEmail(email);
        return new SecurityUser(userAccount);
    }
}

File: back/healthCare/src/main/kotlin/com/healthCare/api/user/GetUserSecurityService.kt
package com.healthCare.api.user

import com.healthCare.api.repository.UserAccountRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service


@Service
class GetUserSecurityService (
    private val userAccountRepository: UserAccountRepository
    ) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(email: String?): UserDetails? {
        val userAccount = userAccountRepository.findByEmail(email!!)
        return SecurityUser(userAccount)
    }
}